python/cuda_parallel/cuda/parallel/experimental/algorithms/__init__.py

```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from .reduce import reduce_into as reduce_into
```

Comment:
I'd try to keep this __init__.py file empty.

From the user perspective: explicit import is better than implicit import.

(I could explain more offline.)

Reply:
> I'd try to keep this init.py file empty.
> From the user perspective: explicit import is better than implicit import.

This isn't necessarily true. For example, as a user, I enjoy being able to do the following:

```python
import pandas as pd

df1 = pd.DataFrame(...)
df2 = pd.DataFrame(...)
result = pd.concat([df1, df2])
```

As opposed to importing things explicitly from where they are defined:

```python
from pandas.core.frame import DataFrame
from pandas.core.reshape.concat import concat

df1 = DataFrame(...)
df2 = DataFrame(...)
result = concat([df1, df2])
```

Most libraries (such as pandas or numpy) expose only a handful of namespaces in their API - and many expose just the top-level namespace. 

This is possible precisely _because_  packages carefully define their  `__init__.py`s to define what they want their public API to look like.

---

Similarly, I think the following is a better API:

```python
from cuda.parallel.experimental.algorithms import (
    reduce,
    for_each,
    transform
)
```

Rather than:

```python
from cuda.parallel.experimental.algorithms.reduce import reduce
from cuda.parallel.experimental.algorithms.for_each import for_each
from cuda.parallel.experimental.algorithms.transform import transform
```

and this needs us to define `algorithms/__init__.py` accordingly.

Reply:
> This isn't necessarily true. For example, as a user, I enjoy

Ack. — Yes, it's a tradeoff.